packages/doenetml-worker-rust/lib-js-wasm-binding/src/math.ts

```diff
@@ -0,0 +1,521 @@
+                ),
+            );
+    } else {
+        return (x: string) =>
+            me.fromAst(
+                new me.converters.latexToAstObj({
+                    appliedFunctionSymbols,
+                    functionSymbols,
+                    allowedLatexSymbols,
+                    parseScientificNotation,
+                }).convert(wrapWordWithVar(x, parseScientificNotation)),
+            );
+    }
+}
+
+export function normalizeLatexString(
+    latexString: string,
+    { unionFromU = false } = {},
+) {
+    let substitutions = [
+        ["\u03B1", "\\alpha "], // 'α'
+        ["\u03B2", "\\beta "], // 'β'
+        ["\u03D0", "\\beta "], // 'ϐ'
+        ["\u0393", "\\Gamma "], // 'Γ'
+        ["\u03B3", "\\gamma "], // 'γ'
+        ["\u0394", "\\Delta "], // 'Δ'
+        ["\u03B4", "\\delta "], // 'δ'
+        ["\u03B5", "\\epsilon "], // 'ε' should this be varepsilon?
+        ["\u03F5", "\\epsilon "], // 'ϵ'
+        ["\u03B6", "\\zeta "], // 'ζ'
+        ["\u03B7", "\\eta "], // 'η'
+        ["\u0398", "\\Theta "], // 'Θ'
+        ["\u03F4", "\\Theta "], // 'ϴ'
+        ["\u03B8", "\\theta "], // 'θ'
+        ["\u1DBF", "\\theta "], // 'ᶿ'
+        ["\u03D1", "\\theta "], // 'ϑ'
+        ["\u03B9", "\\iota "], // 'ι'
+        ["\u03BA", "\\kappa "], // 'κ'
+        ["\u039B", "\\Lambda "], // 'Λ'
+        ["\u03BB", "\\lambda "], // 'λ'
+        ["\u03BC", "\\mu "], // 'μ'
+        ["\u00B5", "\\mu "], // 'µ' should this be micro?
+        ["\u03BD", "\\nu "], // 'ν'
+        ["\u039E", "\\Xi "], // 'Ξ'
+        ["\u03BE", "\\xi "], // 'ξ'
+        ["\u03A0", "\\Pi "], // 'Π'
+        ["\u03C0", "\\pi "], // 'π'
+        ["\u03D6", "\\pi "], // 'ϖ' should this be varpi?
+        ["\u03C1", "\\rho "], // 'ρ'
+        ["\u03F1", "\\rho "], // 'ϱ' should this be varrho?
+        ["\u03A3", "\\Sigma "], // 'Σ'
+        ["\u03C3", "\\sigma "], // 'σ'
+        ["\u03C2", "\\sigma "], // 'ς' should this be varsigma?
+        ["\u03C4", "\\tau "], // 'τ'
+        ["\u03A5", "\\Upsilon "], // 'Υ'
+        ["\u03C5", "\\upsilon "], // 'υ'
+        ["\u03A6", "\\Phi "], // 'Φ'
+        ["\u03C6", "\\phi "], // 'φ' should this be varphi?
+        ["\u03D5", "\\phi "], // 'ϕ'
+        ["\u03A8", "\\Psi "], // 'Ψ'
+        ["\u03C8", "\\psi "], // 'ψ'
+        ["\u03A9", "\\Omega "], // 'Ω'
+        ["\u03C9", "\\omega "], // 'ω'
+        ["\u2212", "-"], // minus sign
+        ["\u22C5", " \\cdot "], // dot operator
+        ["\u00B7", " \\cdot "], // middle dot
+        ["\u222A", " \\cup "], // ∪
+        ["\u2229", " \\cap "], // ∩
+        ["\u221E", " \\infty "], // ∞
+        ["\u2205", " \\emptyset "], // ∅
+        ["\u2032", "'"], // ′
+    ];
+
+    for (let sub of substitutions) {
+        latexString = latexString.replace(new RegExp(sub[0], "g"), sub[1]);
+    }
+
+    let startLdotsMatch = latexString.match(
+        /^(\\ )*(\\ldots|\.(\\ )*\.(\\ )*\.)(\\ )*(.*)$/,
```

Comment:
For complicated regexes, please annotate with some examples of what they're supposed to match.

Reply:
I wish you told that to the version of me from years ago when I wrote this. I'll put in a TODO, as I don't currently remember. Had to do with something about enabling syntax like 1,2,3..., would be my guess.